core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -386,6 +386,16 @@ private static String formatFragment(
         return builder.toString();
     }
 
+    private static TypeProvider getTypeProvider(List<PlanFragment> fragments)
+    {
+        // somewhat faster than java stream
```

Comment:
This is true, but we optimize coordinator code for readability.
While we have a rule not to use Streams in performance sensitive code (like query execution), planner uses streams a lot.

Is this place any special?

Reply:
Ok, I see. I think we could use the streams here.